### ISSUE TITLE
Error code null encountered while starting the PHP process

### ISSUE THREAD

**Issue by enricotedoldi** [adobe/brackets#15291]
### Description

As soon as I open any of my PHP files I receive this error message

<img width="836" alt="Schermata 2020-12-17 alle 14 43 32" src="https://user-images.githubusercontent.com/22967551/102495825-e6f3fb80-4076-11eb-933b-c525ce4c606f.png">

### Versions

OS:
macOS BigSur 11.1
Brackets:
Release 1.14 build di rilascio 1.14.2-17770 (release-1.14.2 f71f00acc) 
build timestamp: Tue Mar 31 2020 00:41:05 GMT-0700

**Comment by NYC-Jim** [comment]
Me too!

**Comment by billb037** [comment]
I'm seeing the same

**Comment by billb037** [comment]
@NYC-Jim I added the same thing as you did and the error was gone after restarting Brackets

@enricotedoldi add the php block that NYC Jim shows in his example to brackets.json, restart Brackets and see if the error disappears

**Comment by enricotedoldi** [comment]
Done, but it gives me the same error I posted

**Comment by billb037** [comment]
@enricotedoldi - yeah, same here. it was fine for the first restart and then came back the next time I opened it. back to square 1 I guess.

**Comment by sidelux** [comment]
Same error here, with no recent changes...

**Comment by sidelux** [comment]
Ok temporary gone after using @NYC-Jim solution but with:
`"enablePhpTooling": false,`
And restarting.

**Comment by billb037** [comment]
@sidelux yeah, changing that value and the error is gone now. Thanks.

**Comment by enricotedoldi** [comment]
ok, but if that is set to "false" you won't have any of these: _Code Hints, Parameter Hints, Linting, Jump To Definition, Find References, Find Document/Project Symbols features._
Am I wrong?

**Comment by asako08** [comment]
I'm getting the same error.
Is it due to updating macOS to Big Sur?
I want to know a better solution.

**Comment by ecortesenv** [comment]
@enricotedoldi Exactly.
@asako08 Probably yes, i think a definitive solution will be available on official update...

**Comment by sidelux** [comment]
@enricotedoldi Exactly.
@asako08 Probably yes, i think a definitive solution will be available on official update...

**Comment by asako08** [comment]
I was able to resolve it by adding executablePath to brackets.json.

**Comment by efhkapp** [comment]
Yes, after added @NYC-Jim solution then add /usr/bin/php to the $PATH and reboot the mac then it works.

For someone who wish to tackle it, can try:
1) Open a .php with Brackets
2) Error comes out, press Open Preferences
3) Copy & Paste @NYC-Jim 's solution into brackets.json
4) Open Terminal, type sudo vi /etc/paths, enter root password
5) Add /usr/bin/php at the end of the file
6) Save and restart mac osx
7) Open .php file again, error should gone.

Thanks all :)

**Comment by theanorak-keith** [comment]
@asako08  You were? I've tried adding the executablePath '/usr/bin/php' to brackets.json (as @NYC-Jim did) but it hasn't solved the error. Did you specifically install PHP separately?

**Comment by asako08** [comment]
@theanorak-keith I use MAMP. So I specified the php in MAMP as executablePath.I think executablePath will be different depending on the environment.

I think the following page will be helpful.
[](https://github.com/adobe/brackets/wiki/PHP-Support-in-Brackets)

**Comment by efhkapp** [comment]
Finally I get it works again by installing another version of PHP, for who wish to tackle it out:

1) install php 7.3 with brew
`brew tap shivammathur/php`
`brew install shivammathur/php/php@7.3`

2) After installed, run
`brew link shivammathur/php/php@7.3`
If error, run `brew doctor`, fix one by one

3) Then
`which php` 
to get a new PHP path, it would like /usr/local/bin/php, copy it and replace in "executablePath" in brackets.json in @NYC-Jim 's solution. 

4) Confirm there is "enablePhpTooling": true setting in  brackets.json

5) Restart Brackets, it should not have any errors, and all the _Code Hints, Parameter Hints, Linting, Jump To Definition, Find References, Find Document/Project Symbols features_ were back

Hope this help. :)

**Comment by sidelux** [comment]
Seems like @efhkapp's solution works.
I have installed php version 7.4 and my executablePath is /usr/local/bin/php.

**Comment by fakerybakery** [comment]
I doubt that Brackets will fix this - they haven't updated it in ages and are now deprecating the project.

**Comment by fakerybakery** [comment]
I noticed that PHP tooling does not work on PHP files, but when I write in HTML files, it does! So strange! I started typing <code>fop</code> and it tooltipped: <code>fopen</code> which is TOTALLY PHP. Wierd!

**Comment by nicolaslule** [comment]
Issued fixed with @NYC-Jim solution. I had to change the value enablePhpTooling": false as recommend by @sidelux. Thanks.
